src/EventJournal.md

```diff
@@ -0,0 +1,75 @@
+
+# Configuring the event journal
+
+### Configuring event journal Capacity
+
+By default, a event journal is configured with a `capacity` of 10000 items. This creates an array with a size of 10000. If 
+a `time-to-live` is configured, then an array of longs is also created that stores the expiration time for every item. 
+In a lot of cases you may want to change this `capacity` number to something that better fits your needs. 
+
+Below is a declarative configuration example of a event journal with a `capacity` of 2000 items.
+
+```xml
+<event-journal enabled="true">
+    <mapName>myMap</mapName>
+    <capacity>5000</capacity>
+    <time-to-live-seconds>20</time-to-live-seconds>
+</event-journal>
+
+<event-journal enabled="true">
+    <cacheName>myCache</cacheName>
+    <capacity>10000</capacity>
+    <time-to-live-seconds>0</time-to-live-seconds>
+</event-journal>
+``` 
+You can also configure an event journal programmatically. The following is a programmatic version of the above declarative configuration. 
+
+```java
+EventJournalConfig myMapJournalConfig = new EventJournalConfig()
+                .setMapName("myMap")
+                .setEnabled(true)
+                .setCapacity(5000)
+                .setTimeToLiveSeconds(20);
+
+        EventJournalConfig myCacheJournalConfig = new EventJournalConfig()
+                .setMapName("myCache")
+                .setEnabled(true)
+                .setCapacity(10000)
+                .setTimeToLiveSeconds(0);
+        
+        Config config = new Config();
+        config.addEventJournalConfig(myMapJournalConfig);
+        config.addEventJournalConfig(myCacheJournalConfig);
+```
+
+
+The `mapName` and `cacheName` attributes define the map or cache to which this event journal configuration applies. You can use pattern-matching and the `default` keyword when doing so. For instance, by using a `mapName` of `journaled*`, the journal configuration will apply to all maps whose names start with "journaled" and don't have other journal configurations that match (e.g. if you would have a more specific journal config with an exact name match). If you specify the `mapName` or `cacheName` as `default`, the journal configuration will apply to all maps and caches that don't have any other journal configuration. This means that potentially all maps and/or caches will have one single event journal configuration.
+
+
+### Event journal partitioning
+
+The event journal is a partitioned data structure. The partitioning is done by the event key. Because of this, the map and cache entry with a specific key is co-located with the events for that key and will be migrated accordingly.
+Also, the backup count for the event journal is equal to the backup count of the map or cache for which it contains events. The events on the backup replicas will be created with the map or cache backup operations and no additional network traffic is introduced when appending events to the event journal. 
```

Comment:
`The events on the backup replicas will be created with the map or cache backup operations` afaik if you configure eviction on map, primary and backup may evict different entries and may diverge. this will affect the event journal I guess ?